tools/cellxgene_census_builder/src/cellxgene_census_builder/__main__.py

```diff
@@ -38,21 +39,71 @@ def main() -> int:
     # Process initialization/setup must be done early. NOTE: do NOT log before this line!
     process_init(build_args)
 
-    # Return process exit code (or raise, which exits with a code of `1`)
-    return do_build(build_args, skip_completed_steps=cli_args.test_resume)
+    command = os.getenv("CENSUS_BUILD_COMMAND")
+    if command == "build":
+        return do_build(build_args)
+    elif command == "replicate":
+        return do_replicate(build_args)
+    elif command == "release":
+        return do_release(build_args)
+    elif command == "sync-release":
+        return do_sync_release(build_args)
+    elif command == "pass":
+        return 0  # Used for testing
+    else:
+        return do_full_build(build_args, skip_completed_steps=cli_args.test_resume)
 
 
-def do_build(args: CensusBuildArgs, skip_completed_steps: bool = False) -> int:
+def do_build(args: CensusBuildArgs) -> int:
     """
-    Top-level build sequence.
-
-    Built steps will be executed in order. Build will stop if a build step returns non-zero
-    exit code or raises.
+    Performs a build of the census and stores the artifact on the primary bucket.
+    No release or cleanup is performed.
     """
     logging.info(f"Census build: start [version={__version__}]")
     logging.info(args)
+    build_steps = [
+        do_prebuild_set_defaults,
+        do_prebuild_checks,
+        do_build_soma,
+        do_validate_soma,
+        do_create_reports,
+        do_data_copy,
+        do_report_copy,
+        do_log_copy,
+    ]
+    rv = _do_build(build_steps, args)
+    logging.info("Census build: completed")
+    return rv
+
 
-    build_steps: List[Callable[[CensusBuildArgs], bool]] = [
+def do_release(args: CensusBuildArgs) -> int:
```

Comment:
I would prefer that we have a naming convention that differentiates between the single-step "do something" functions (e.g., do_prebuild_checks), and the top-level UX functions which execute _groups_ of sub-steps.

A possible scheme is to call the aggregate UX functions "groups", so rename do_build to do_build_group, etc.

Just trying for clarity / comprehensibility.  So feel free to improve/modify

Reply:
Also, any reason the three top-level UX commands (release, build, replicate) are not just defined by a data structure containing their steps, with a generic driver, e.g.,
```
{
    "release": [do_the_release],
    "replicate": [do_sync_artifact_to_replica_s3_bucket],
    "build": [
        do_prebuild_set_defaults,
        do_prebuild_checks,
        do_build_soma,
        do_validate_soma,
        do_create_reports,
        do_data_copy,
        do_report_copy,
        do_log_copy,
    ]
}
```